include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -406,11 +405,11 @@ __kernel_work_group_size(_ExecutionPolicy&& __policy, const sycl::kernel& __kern
 }
 
 template <typename _ExecutionPolicy>
-long
+::std::size_t
 __kernel_sub_group_size(_ExecutionPolicy&& __policy, const sycl::kernel& __kernel)
 {
-    auto __device = __policy.queue().get_device();
-    auto __wg_size = __kernel_work_group_size(::std::forward<_ExecutionPolicy>(__policy), __kernel);
+    const sycl::device& __device = __policy.queue().get_device();
+    const ::std::size_t __wg_size = __kernel_work_group_size(::std::forward<_ExecutionPolicy>(__policy), __kernel);
     const ::std::size_t __sg_size =
 #if _USE_KERNEL_DEVICE_SPECIFIC_API
         __kernel.template get_info<sycl::info::kernel_device_specific::max_sub_group_size>(
```

Comment:
According to the SYCL 2020 it returns `uint32_t`, not `size_t`. Please, also align types where the function is calling from

Reply:
I changed the type used in the function, but left the types where it is called, because `max_sub_group_size` and `sub_group_sizes` are used interchangeably (depending on presence of `_ONEDPL_COMPILE_KERNEL` macro). `sub_group_sizes` provides `size_t` type for a specific size, which has equal or bigger width compared with `uint32_t` on the supported systems.